packages/next-swc/crates/core/src/server_actions.rs

```diff
@@ -338,22 +361,89 @@ impl<C: Comments> VisitMut for ServerActions<C> {
         let old_annotations = self.annotations.take();
 
         let mut new = Vec::with_capacity(stmts.len());
+
+        // We need a second pass to collect all async function idents and exports
+        // so we can handle the named export cases if it's in the "use server" file.
+        if self.in_action_file {
+            self.in_prepass = true;
+            for stmt in stmts.iter_mut() {
+                match &*stmt {
+                    ModuleItem::ModuleDecl(ModuleDecl::ExportDecl(ExportDecl {
+                        decl: Decl::Var(var),
+                        ..
+                    })) => {
+                        for decl in &var.decls {
+                            if let Pat::Ident(ident) = &decl.name {
+                                // export var foo = 1;
+                                self.exported_idents.push(ident.id.to_id());
+                            }
+                        }
+                    }
+                    ModuleItem::ModuleDecl(ModuleDecl::ExportNamed(named)) => {
```

Comment:
We should ignore this node if `named.src` is not `None`.

Reply:
It's a named export like

```ts
export { foo as bar } from './foo';
```

Reply:
But I'm not sure what should this do in case of such re-exports

Reply:
I think we have to ignore re-exports and do runtime check for them. Good point!